### PR TITLE
fix(chat): fix infinity loader when try to use search on not allowed folder, fix items to review counter (Issues #1667, #1842)

### PR DESCRIPTION
**Pull request by Alexander-Kezik**
**Description:**

- fix infinity loader when try to use search on not allowed folder
- fix items to review counter

Issues:

- #1667
- #1842

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1845.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx [hunk 1]
```diff
@@ -232,8 +232,15 @@ export const ApproveRequiredSection = ({
...
       additionalNode={
-        publicationsToReviewCount && (
-          <span className="absolute right-4 flex h-[14px] select-none items-center justify-center rounded bg-accent-secondary px-[2px] text-[10px] font-semibold text-controls-disable">
+        !!publicationsToReviewCount && (
+          <span
+            className={classNames(
+              'absolute right-4 flex h-[14px] min-w-[14px] select-none items-center justify-center rounded px-[2px] text-[10px] font-semibold text-controls-disable',
+              featureTypes.includes(FeatureType.Chat)
+                ? 'bg-accent-secondary'
+                : 'bg-accent-tertiary',
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/Publish/ApproveRequiredSection.tsx]: Just use `bg-accent-primary` instead. We shouldn't adapt primary color for panels - it happens automatically

File: apps/chat/src/store/ui/ui.reducers.ts [hunk 2]
```diff
@@ -159,13 +159,22 @@ export const uiSlice = createSlice({
...
+        featureType: FeatureType;
+        options?: Partial<{ doNotAutoOpenFolders: boolean }>;
+      }>,
     ) => {
-      const featureType = payload.featureType;
-      const openedFoldersIds = state.openedFoldersIds[featureType];
-      const isOpened = openedFoldersIds.includes(payload.id);
-      if (!isOpened) {
-        state.openedFoldersIds[featureType].push(payload.id);
+      if (!payload?.options?.doNotAutoOpenFolders) {
```
Comment (IlyaBondar) [inline apps/chat/src/store/ui/ui.reducers.ts]: action `openFolder`, but don't add folder id in `openedFoldersIds`? What happens here?

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1845.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/store/conversations/conversations.epics.ts [hunk 3]
```diff
@@ -2424,7 +2446,14 @@ const uploadConversationsWithFoldersRecursiveEpic: AppEpic = (
                 of(
                   ConversationsActions.uploadChildConversationsWithFoldersSuccess(
                     {
-                      parentIds: [...payload.path, ...paths],
+                      parentIds: [
+                        ...folders
+                          .filter((folder) =>
+                            folder.id.startsWith(payload.path || ''),
+                          )
+                          .map((folder) => folder.id),
```
Comment (IlyaBondar) [inline apps/chat/src/store/conversations/conversations.epics.ts]:
```suggestion
                        ...(payload.path ? folders
                          .filter((folder) =>
                            folder.id.startsWith(payload.path),
                          ) : folders)
                          .map((folder) => folder.id),
```

File: apps/chat/src/store/prompts/prompts.epics.ts [hunk 4]
```diff
@@ -599,7 +592,14 @@ const uploadPromptsWithFoldersRecursiveEpic: AppEpic = (action$, state$) =>
               concat(
                 of(
                   PromptsActions.uploadChildPromptsWithFoldersSuccess({
-                    parentIds: [...payload.path, ...paths],
+                    parentIds: [
+                      ...folders
+                        .filter((folder) =>
+                          folder.id.startsWith(payload.path || ''),
+                        )
+                        .map((folder) => folder.id),
```
Comment (IlyaBondar) [inline apps/chat/src/store/prompts/prompts.epics.ts]:
```suggestion
                      ...(payload.path ? folders
                        .filter((folder) =>
                          folder.id.startsWith(payload.path),
                        ): folders)
                        .map((folder) => folder.id),
```

File: apps/chat/src/store/conversations/conversations.epics.ts (hunk 3 above)
Comment (Alexander-Kezik) [reply apps/chat/src/store/conversations/conversations.epics.ts]: reverted

File: apps/chat/src/store/prompts/prompts.epics.ts (hunk 4 above)
Comment (Alexander-Kezik) [reply apps/chat/src/store/prompts/prompts.epics.ts]: reverted

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1845.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by Alexander-Kezik** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1845.nightly-test.deltixhub.io
>E2E tests status: success

**Review by IlyaBondar — APPROVED** [review]

**Pull request merged**